lib/spack/spack/environment/environment.py

```diff
@@ -100,7 +101,7 @@ def default_manifest_yaml():
 valid_environment_name_re = r"^\w[\w-]*$"
 
 #: version of the lockfile format. Must increase monotonically.
-lockfile_format_version = 4
+lockfile_format_version = 5
```

Comment:
I am not sure we need to increment this.  I wonder if we should have major and minor lock file versions, because the addition of the `spack` section doesn't actually prevent prior versions of Spack from reading this file.

Unfortunately, changing this to `"4.1"` would prevent prior spack versions from reading the lockfile version... so we should either just make a clean break and go with, say, `"5.0"` or keep it at 4 and support the spack section optionally.

Reply:
I like that latter.